litellm/llms/together_ai/chat.py

```diff
@@ -6,8 +6,54 @@
 Docs: https://docs.together.ai/reference/completions-1
 """
 
+from typing import Optional
+
+from litellm import get_model_info, verbose_logger
+
 from ..OpenAI.chat.gpt_transformation import OpenAIGPTConfig
 
 
 class TogetherAIConfig(OpenAIGPTConfig):
-    pass
+    def get_supported_openai_params(self, model: str) -> list:
+        """
+        Only some together models support response_format / tool calling
+
+        Docs: https://docs.together.ai/docs/json-mode
+        """
+        supports_function_calling: Optional[bool] = None
+        try:
+            model_info = get_model_info(model, custom_llm_provider="together_ai")
+            supports_function_calling = model_info.get(
+                "supports_function_calling", False
+            )
+        except Exception as e:
+            verbose_logger.debug(f"Error getting supported openai params: {e}")
+            pass
+
+        optional_params = super().get_supported_openai_params(model)
+        if supports_function_calling is not True:
+            verbose_logger.warning(
+                "Only some together models support function calling/response_format. Docs - https://docs.together.ai/docs/function-calling"
+            )
+            optional_params.remove("tools")
+            optional_params.remove("tool_choice")
+            optional_params.remove("function_call")
+            optional_params.remove("response_format")
+        return optional_params
```

Comment:
@krrishdholakia can you explain why we shouldn't add similar code for supports_response_schema?